apps/site/lib/site_web/views/trip_plan_view.ex

```diff
@@ -607,20 +607,30 @@ defmodule SiteWeb.TripPlanView do
           round_trip_total: Format.price(one_way_total_fare * 2)
         )
 
-      fares = get_calculated_fares(i)
+      # If Logan airport is the origin, all subsequent subway trips from there should be free
+      first_transit_leg = i |> Itinerary.transit_legs() |> List.first()
+
+      readjusted_itinerary =
+        if stop_is_silver_line_airport?([first_transit_leg], :from) do
```

Comment:
It looks like `readjust_itinerary_with_free_fares` could use some tests. But also, could this logic be done when fares are initially calculated in the controller? It seems like it would be better if the view didn't have to update the data.

Reply:
You're right. I also "relocated" some functions that (I felt like) belonged to other modules (e.g. Leg, Fares) rather than to the view. Let me know if these latest changes make sense.